ansible_base/lib/dynamic_config/settings_logic.py

```diff
@@ -0,0 +1,255 @@
+from copy import copy
+from typing import Optional
+
+#
+# If you are adding a new dynamic setting:
+#     Please be sure to modify pyproject.toml with your new settings in tool.setuptools.dynamic
+#     Add a new requirements/requirements_<section>.in /even if its an empty file/
+#
+
+
+def get_dab_settings(
+    installed_apps: list[str],
+    rest_framework: Optional[dict] = None,
+    spectacular_settings: Optional[dict] = None,
+    authentication_backends: Optional[list[str]] = None,
+    middleware: Optional[list[str]] = None,
+    oauth2_provider: Optional[dict] = None,
+) -> dict:
+    dab_data = {}
+
+    # The org and team abstract models cause errors if not set, even if not used
+    dab_data['ANSIBLE_BASE_TEAM_MODEL'] = 'auth.Group'
+    dab_data['ANSIBLE_BASE_ORGANIZATION_MODEL'] = 'auth.Group'
```

Comment:
This is already set on galaxy `settings` https://github.com/ansible/galaxy_ng/blob/master/galaxy_ng/app/settings.py#L390

So if we call `get_dab_settings` we need to ensure the existing variables are not overridden, and when the existing variable is a data structure like a `dict` or `list` we need to ensure it has the values merged.

On galaxy the merging is controlled by the merging markers.

What if

```python
def get_dab_settings(
    ...
    dab_data: Optional[dict] = None
): -> dict:
    if dab_data is None:
        dab_data = {}

    dab_data.setdefault("ANSIBLE_BASE_TEAM_MODEL", "auth.group")
    ...

    return dab_data
```

This does not solve the merging part, but at least allows us to pass a prepopulated `dab_data` that will be changed only if defaults doesn't exist yet.

That would also prevent override https://github.com/ansible/galaxy_ng/blob/master/galaxy_ng/app/settings.py#L226 and other variables like that one.

And for the merging parts we try to resolve on dynaconf side.

For authentication backends for example we cant simply override from DAB because we have very specific logic for that https://github.com/ansible/galaxy_ng/blob/b26816eaf9c1e34f4033835e09125013a5d182d3/galaxy_ng/app/settings.py#L333-L349

Reply:
My intent is that `get_dab_settings` is a dictionary that _only represents_ default values to begin with. If you look at the dynamic_settings module, it skips any variables that are defined locally.

The current behavior is already fairly wrong/inconsistent on this point. For some variables, we reference it and only set it when getting a `NameError` exception, but for others we don't. So if you set a variable before doing the include, the dynamic_settings (currently, in devel) will overwrite it. This formalizes the practice so that it won't overwrite a user setting even if you set it before doing the include.

I would really prefer that `get_dab_settings` does not have knowledge of the entire settings. Because I want DAB developers to have to declare what variables from the user's settings it needs to see. I don't want DAB to be able to reference `dab_data["SOME_NEW_KEY"]` so that they get the value of SOME_NEW_KEY from the user settings. If that need does arise, that's fine, but it's a notable changelog-worthy kind of thing.

Reply:
Ok, so after getting the `dab_data` we can get only the complement keys to populate galaxy settings.